dataprofiler/tests/data_readers/test_csv_data.py

```diff
@@ -317,29 +319,35 @@ def test_header_check_files(self):
         Determine if files with no header are properly determined.
         """
         from itertools import islice
-        from dataprofiler.data_readers import data_utils
 
         # add some more files to the list to test the header detection
         # these files have some first lines which are not the header
         for input_file in self.input_file_names:
-            with open(input_file['path'], encoding=input_file['encoding']) as csvfile:
+            with open(input_file['path'],
+                      encoding=input_file['encoding']) as csvfile:
                 data_as_str = ''.join(list(islice(csvfile, 5)))
-            header_line = CSVData._guess_header_row(data_as_str, input_file['delimiter'])
-            self.assertIn(header_line, input_file['has_header'], input_file['path'])
+            header_line = CSVData._guess_header_row(data_as_str,
+                                                    input_file['delimiter'])
+            self.assertIn(header_line, input_file['has_header'],
+                          input_file['path'])
 
         for input_buf in self.buffer_list:
             # BytesIO is wrapped so that it is fed into guess header row 
             # the same way it would internally
+            buffer = input_buf['path']
             if isinstance(input_buf['path'], BytesIO):
-                input_buf['path'] = TextIOWrapper(input_buf['path'], encoding=input_buf['encoding'])
```

Comment:
this was a bug replacing the value in the dict as oppose to maintaining the BytesIO.